packages/core/core/src/public/Asset.js

```diff
@@ -196,6 +196,10 @@ export class Asset extends BaseAsset implements IAsset {
   get stats(): Stats {
     return this.#asset.value.stats;
   }
+
+  get publicId(): string {
```

Comment:
Should we expose `asset.id` at all anymore? Should `asset.id` be the public id?

Reply:
Unfortunately we need the entire asset graph to determine asset public ids, as they may need to be extended in length to mitigate collisions. This means that during transformation, these public ids won't be available, as the graph is not yet complete and we can't independently generate these shorter ids. Asset ids are used in at least hoisting in JS transformation.

I'd like to use public ids for the identifiers we create in scope hoisting, but unfortunately we need them in hoisting (transformation). We could insert placeholders and replace them during `concat` or `link` in packaging. What do you think, @devongovett @mischnic?